recipes/fast-dds/all/conanfile.py

```diff
@@ -0,0 +1,209 @@
+    _cmake = None
+    exports_sources = ["patches/**", "CMakeLists.txt"]
+
+    @property
+    def _pkg_share(self):
+        return os.path.join(
+            self.package_folder,
+            "share"
+        )
+
+    @property
+    def _pkg_tools(self):
+        return os.path.join(
+            self.package_folder,
+            "tools"
+        )
+
+    @property
+    def _pkg_bin(self):
+        return os.path.join(
+            self.package_folder,
+            "bin"
+        )
+
+    @property
+    def _module_subfolder(self):
+        return os.path.join(
+            "lib",
+            "cmake"
+        )
+
+    @property
+    def _module_file_rel_path(self):
+        return os.path.join(
+            self._module_subfolder,
+            "conan-target-properties.cmake"
+        )
+
+    @staticmethod
+    def _create_cmake_module_alias_targets(module_file, targets):
+        content = ""
+        for alias, aliased in targets.items():
+            content += textwrap.dedent("""\
+                if(TARGET {aliased} AND NOT TARGET {alias})
+                    add_library({alias} INTERFACE IMPORTED)
+                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+                endif()
+            """.format(alias=alias, aliased=aliased))
+        tools.save(module_file, content)
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)         
+            self._cmake.definitions["BUILD_MEMORY_TOOLS"] = False
+            self._cmake.definitions["NO_TLS"] = not self.options.with_ssl
+            self._cmake.definitions["SECURITY"] = self.options.with_ssl
+            self._cmake.definitions["EPROSIMA_INSTALLER_MINION"] = False
+            self._cmake.configure()
+        return self._cmake
+
+    def requirements(self):
+        self.requires("tinyxml2/7.1.0")
+        self.requires("asio/1.18.2")
+        self.requires("fast-cdr/1.0.21")
+        self.requires("foonathan-memory/0.7.0")
+        if self.options.with_ssl:
+            self.requires("openssl/1.1.1k")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True,
+                  destination=self._source_subfolder)
+
+    def validate(self):
+        os = self.settings.os
+        compiler = self.settings.compiler
+        version = tools.Version(self.settings.compiler.version)
+        if compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 11)
+        if os == "Linux" and compiler == "gcc" and version < "5":
```

Comment:
We have a template conan-io/conan#8002 you can reuse to also print a warning for the consumer

Reply:
got your point - i took the dictionary for C++11 from https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/cxxopts/all/conanfile.py.

Reply:
and btw - nice idea - to have a standard template here ;)